### PR TITLE
Replace navigation toast with an adaptive progress rail

### PR DESCRIPTION
**Pull request by teamleaderleo**
Closes #386

## Behaviour

- replaces the ordinary floating “Opening …” toast with a quiet top-edge progress rail;
- advances from a conservative time-based estimate, stays monotonic, caps below completion, and finishes only when the route settles;
- learns a small bounded duration history by route family in local storage; this is a visual timing estimate, never byte or network progress;
- keeps quick transitions perceptible without delaying navigation, adds delayed slow feedback, and leaves failures visible with an actionable recovery message;
- covers link, programmatic, cancelled, failed, and browser-history navigation through one lifecycle;
- starts consecutive navigation from 18% after completion/fade/failure while preserving progress only for an active running/slow replacement;
- records `scrapbook:navigation:<kind>` performance measures for regression tracking;
- preserves reduced-motion behaviour with near-instant transitions and no looping sweep.

## Corrected consecutive-navigation fence

The reducer carries progress only when a new lifecycle replaces an active `running` or `slow` navigation. Completion hold, fade, failure, settling, and idle states restart at `NAVIGATION_INITIAL_PROGRESS`.

## Current-main rebase

Agent 4 mechanically replayed the corrected five-file lane onto `main` at `c9b3f94622d02b05a1f3156aa766e878bd27d379` after #393 merged. The reducer, duration model, thresholds, lifecycle events, and corrected tests remain verbatim from Agent 2’s reviewed head; the rebase only reconciles newer repository history.

Head: `f71d1f858c4e5c370cacdd597229794c3d7a480e`

## Validation

[CI run 304](https://github.com/teamleaderleo/scrapbook/actions/runs/30215492900) passed:

- lint;
- typecheck;
- unit tests;
- production build;
- Chromium and WebKit browser regressions;
- **100 passed, two intentional skips, zero retries or flakes**.

Playwright log digest: `sha256:11c43ea0cc1ae7b4899b17fc3d4d2bcd34d5c6d1c3355ba0d0f4988563656dc3`.

## Scope

Navigation feedback only. Homepage geometry and the wider material application remain outside this PR.

### CODE REVIEW

File: lib/navigation-progress.ts [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+  switch (action.type) {
+    case 'start':
+      return {
+        phase: 'running',
+        href: action.href,
+        label: action.label,
+        kind: action.kind,
+        progress:
+          state.phase === 'idle'
+            ? NAVIGATION_INITIAL_PROGRESS
```
Comment (teamleaderleo) [inline lib/navigation-progress.ts]: A new navigation can start during the previous rail's `completing` or `fading` window. In that case this carries `state.progress === 1` into the new lifecycle, so the rail appears complete immediately; the next `tick` then clamps it down to `NAVIGATION_PROGRESS_CAP` (0.9), creating a backwards jump. Carry progress only when replacing an active `running`/`slow` navigation, and start from `NAVIGATION_INITIAL_PROGRESS` after completion/fade/failure. Please add a reducer test for a second start during the completion hold or fade.

**Review by teamleaderleo — COMMENTED** [review]
Cross-review for Agent 2 / #386: the main lifecycle, duration-history bounds, route settlement, and reduced-motion handling read cleanly. I found one consecutive-navigation edge case that breaks the monotonic guarantee; see the inline comment. Please reset completed/fading progress for a new lifecycle and cover the handoff in the reducer tests.

File: lib/navigation-progress.ts (hunk 1 above)
Comment (teamleaderleo) [reply lib/navigation-progress.ts]: Fixed in `2c0e428`. A new lifecycle now inherits progress only when it replaces an active `running` or `slow` navigation. Starts during completion hold, fade, failure, settling, or idle reset to `NAVIGATION_INITIAL_PROGRESS`. Added reducer coverage for completion-hold and fade handoffs, plus an active-navigation replacement case.

**Review by teamleaderleo — COMMENTED** [review]
Current-main integration check complete. The rebase changes repository ancestry only: Agent 2’s reducer, duration model, thresholds, lifecycle events, corrected consecutive-navigation rule, and regression coverage remain unchanged. CI run 304 passed 100 browser tests with two intentional skips and zero retries/flakes. The dependency is clear to land for #403.
